### DISCUSSION TITLE
Filter Timeline events by event type

### DISCUSSION THREAD

**Discussion by eneiss** [mealie-recipes/mealie#3250] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

It is currently impossible to filter [timeline](https://demo.mealie.io/g/home/recipes/timeline) events.
This can be annoying because you might want to see only "I made this!" events and ignore recipe creation ones.

**It is especially true when migrating to 1.x from 0.x**, as every imported recipe creates its own event in the timeline, flooding it with false creation events.

### Please provide a concise description of the feature that would resolve your issue.

I would like something like a button on the [Timeline](https://demo.mealie.io/g/home/recipes/timeline) page (next to the sort one) that could show a list of event types with associated checkboxes.
This list could be something like:

Show events of type:
- [x] I made this
- [x] Recipe created
- [x] ... (I don't know if there are other existing event types)

Everything would be checked by default, but you could uncheck event types to hide their corresponding events from the timeline.

I'm not a UI pro so there might be better solutions, that's just my two cents 🙂 

### Please consider and list out some caveats or tradeoffs made in your design decision

It would add minor cluttering to the UI, but I believe it would be fine if it were a menu hidden behind a button or something discreet.

Maybe the checkbox list is not the best UI choice either, I'll leave it to whoever wants to take care of this feature to follow their own opinion on how to do it best :)

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [ ] If this is accepted I'm willing to help maintain this feature
- [ ] I'm willing to sponsor/pay a developer to do this work

**Comment by michael-genson** [comment]
Implemented in https://github.com/mealie-recipes/mealie/pull/3284, thanks for the suggestion!

**Reply by eneiss** [reply]
Thanks for implementing it so fast!! 🚀
